### ISSUE TITLE
Backport changes from #11 to `holo-nixpkgs/hydra`

### ISSUE THREAD

**Issue by firestack** [Holo-Host/bugs-n-issues#35]
https://github.com/Holo-Host/bugs-n-issues/issues/11 has been deployed to `devNet` but it has not been deployed to `hydra.holo.host`.

### Merges to backport
- https://github.com/Holo-Host/holo-nixpkgs/pull/1140
- https://github.com/Holo-Host/holo-nixpkgs/pull/1162

**Comment by peeech** [comment]
Yes, currently hydra is following a branch `hydra`. My intention is to rebase `hydra` on develop.

**Comment by cole-h** [comment]
I think it may be necessary to also have `develop` pull in `hydra`'s changes. https://github.com/Holo-Host/holo-nixpkgs/pull/1092 is only present on the `hydra` branch, which means only PRs that are based on `hydra` have the fixed constituents logic (this is my bad for not backporting, sorry about that).

Do you want to handle that as well?

**Comment by peeech** [comment]
Yes I will. I have noticed today on a diff `develop..hydra` that this is a case and I think the best thing to do is to merge `hydra` (after #1173 merge) into `develop`.

This actually leads me to a dilemma: is it worth keeping separate branch for hydra? An obvious up-side of using develop for Hydra is eliminating situations like this one. The only thing I am afraid of is that Hydra is actually an important element of our production environment. We do not really have any tests showing that Hydra won't be broken by future merges of code into develop. For holoports we have a nightly-test suite. For Hydra we have nada.

I am curious to hear your opinions here @firestack , @cole-h , @grahamc

**Comment by cole-h** [comment]
I noticed that https://github.com/Holo-Host/holo-nixpkgs/pull/1173 was reverted (in https://github.com/Holo-Host/holo-nixpkgs/pull/1181) -- is there any particular reason why this happened?

**Comment by peeech** [comment]
I created PR which addresses an issue of https://github.com/Holo-Host/holo-nixpkgs/pull/1092 not being merged into develop.

Now https://github.com/Holo-Host/holo-nixpkgs/pull/1173 was reverted because it broke hydra. This is a good example of a flaw in our branch management in holo-nixpkgs. Merges should be straightforward and propagate to develop and master without having to do any additional work other than merges and releases. We are discussing changes internally and will update everyone interested soon.
